lib/web_ui/lib/src/engine/canvaskit/texture.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/ui.dart' as ui;
+
+import '../texture.dart';
+import 'canvaskit_api.dart';
+import 'image.dart';
+import 'layer.dart';
+import 'painting.dart';
+
+class CkTexture {
+  CkTexture(this.source);
+
+  final Object source;
+
+  CkImage? _ckImage;
+
+  bool _newFrameReady = false;
+
+  void markNewFrameAvailable() {
+    _newFrameReady = true;
+  }
+
+  void paint(
+    PaintContext context,
+    ui.Offset offset,
+    double width,
+    double height,
+    bool freeze,
+    ui.FilterQuality filterQuality,
+  ) {
+    if (_ckImage == null || _newFrameReady && !freeze) {
+      final SkImage? skImage =
+          canvasKit.MakeLazyImageFromTextureSource(source, null);
+
+      if (skImage == null) {
+        return;
+      }
+
+      _ckImage?.dispose();
+      _ckImage = CkImage(skImage);
+    }
+
+    context.leafNodesCanvas?.drawImageRect(
+      _ckImage!.clone(),
+      ui.Rect.fromLTWH(
+        0,
+        0,
+        _ckImage!.width.toDouble(),
+        _ckImage!.height.toDouble(),
+      ),
+      ui.Rect.fromLTWH(offset.dx, offset.dy, width, height),
+      CkPaint()..filterQuality = filterQuality,
```

Comment:
`CkPaint` objects tend to be expensive. If all we care about is `filterQuality`, we can store `CkPaint` in a field and update the `filterQuality` field as needed.